aten/src/ATen/native/cpu/int4mm_kernel.cpp

```diff
@@ -936,7 +936,7 @@ void dyn_quant_pack_4bit_weight_kernel(
 #if AT_KLEIDIAI_ENABLED()
   if (can_use_kleidiai(scales_zeros, K, block_size)) {
     const int64_t weight_packed_size =
-        kleidiai::kai_pack_rhs_int4_size(N, K, block_size);
+        kleidiai::kai_pack_rhs_int4_size(N, K, block_size, weights.scalar_type());
```

Comment:
you want to pass `inp.scalar_type()` [here](https://github.com/pytorch/pytorch/pull/158250/files#diff-3dcbe3f22985e9eb2b5ce8fbfb3c7222cef9a7ad1de4a44494995c98d4b4f77cR1282:~:text=const%20int64_t%20weight_packed_size%20%3D-,kleidiai%3A%3Akai_pack_rhs_int4_size(N%2C%20K%2C%20block_size)%3B,-if%20(weight_packed_size%20%3D%3D%20packed_weights) as well